docs/src/development/linux.md

```diff
@@ -70,10 +70,31 @@ cargo test --workspace
 
 Zed has basic support for both modes. The mode is selected at runtime. If you're on wayland and want to run in X11 mode, you can set `WAYLAND_DISPLAY='' cargo run` to do so.
 
+## Notes for packaging Zed
+
+Thank you for taking on the task of packaging Zed! There are a few things to note to make things work smoothly:
+
+* Zed is a *very* fast moving project. We typically ship 2-3 [releases per week](/docs/development/releases) to fix user-reported issues and major new features. Please make sure that you are packaging in a way where updates will be available to users within a few hours or days.
+* If at all possible we recommend starting from the tarball that we [build](https://github.com/zed-industries/zed/releases) and release (currently only for "preview" versions) rather than building your own copy. This ensures you're running the same code and configuration that we support and test.
+* If that is not possible, read through [script/bundle-linux](https://github.com/zed-industries/zed/blob/main/script/bundle-linux) to see what is involved.
+
+The minimal technical requirements are (probably) that:
+* You will need to take the built package for `crates/cli` and put it in `$PATH` with the name `zed`.
+* You will need to take the built binary for `crates/zed` and put it at `$PATH/to/cli/../../libexec/zed-editor`. For example if you are going to put the cli at `~/.local/bin/zed` put zed at `~/.local/libexec/zed-editor`.
+* You will need to ensure that the necessary libraries are installed. You can get the current list from the lib directory of the tarball we provide, or by [inspecting the built binary](https://github.com/zed-industries/zed/blob/059a4141b756cf4afac4c977afc488539aec6470/script/bundle-linux#L65-L70) on your system.
+
+Also, beware! Zed does a number of things that are required to build a good IDE, but which are a bit "out there" from the point of view of package managers and linux distributions:
+* We automatically install the correct version of developer tools. Like rustup/rbenv/pyenv, etc. we've found that "one system-wide version" does not work well for many professional development environments. Users expect us to support multiple projects running on multiple versions of the toolchain out of the box.
+* We allow users to install extensions which may install further tooling as needed.
```

Comment:
Which may be fine **if** the user is properly informed and can decide whether or not to proceed. This information must include **at least** names and licences of all tools to be installed.

Reply:
I made a comment https://github.com/zed-industries/zed/issues/12589#issuecomment-2148425181 which would be a nice way to deal with this. Basically you let the user install the extension, but if it wants to download anything not already on the system, it has to ask for permission (which would work pretty easily with the extension system because it already has to go through a [function](https://docs.rs/zed_extension_api/0.0.6/zed_extension_api/fn.download_file.html) to download anything).